### ISSUE TITLE
Browser navigation to known room alias always resolves room ID

### ISSUE THREAD

**Issue by ofek** [vector-im/riot-web#10020]
### Description

First off, awesome project!

Due to my disability I must use an on-screen keyboard for everything. As such, I try to take full advantage of shortcut keys to limit the number of keystrokes.

My standard workflow in Slack is that I open/click all channels I expect to use in order of frequency and then navigate solely via the `BrowserBack` and `BrowserForward` buttons (since clicking all around is quite strenuous for me).

Buttons: https://docs.microsoft.com/en-us/windows-hardware/customize/enterprise/predefined-key-combinations#browser-keys
Pic of my keyboard options:

![Capture](https://user-images.githubusercontent.com/9677399/59325340-b1679880-8cb0-11e9-820b-26825c447768.PNG)

This works in Riot, Visual Studio Code, and other Electron apps too.

However, in Riot doing so will trigger a refresh of the room, which takes ~2-5 seconds it seems.

Any idea why it is different than clicking on a room?

### Version information

- **Platform**: desktop
- **OS**: Windows
- **Version**: 1.2.1

**Comment by jryans** [comment]
Thanks for filing! You are correct that browser back / forward is slower than the room list, but as far as I know, that's a bug and not intentional.

**Comment by jryans** [comment]
Ah, the issue is about room aliases vs. room IDs: if the room has an alias, we currently resolve the alias to an ID every time when switching rooms via hash change / back / forward buttons, whereas the room list always changes by room ID only.

**Comment by ofek** [comment]
Oh interesting! Would that be easy to fix?

**Comment by jryans** [comment]
I think we'd need a bit somewhat careful because it is possible to retarget a room alias from one room to another, so that's an edge case we need to think through here.

We could consider things like:

* Use `pushState` etc. when switching rooms so the room ID can "ride along" with the alias even if not in the location bar
* Keep a room alias to ID cache with some kind of TTL (a few hours, say)

These approaches should keep room switching speedy without much risk of going to the wrong place (because an alias changed targets).

**Comment by ofek** [comment]
Any update?

**Comment by jryans** [comment]
If there's an update, it will appear here. At the moment, this is in a backlog of things we'd like to fix, but there are also many other things in the same list.

**Comment by jryans** [comment]
I'm going to take this on. It really contributes to Riot feeling sluggish for no good reason.

**Comment by ofek** [comment]
Thanks!
